### PR TITLE
#84: InitialsBuilder inside 3DB support

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-compose/issues/84 |
| Dependencies | https://github.com/ripe-tech/ripe-sdk/pull/205 (MERGED) |
| Decisions | • Function to connect to the endpoint that returns the logic `js` script for the build. <br> • Get the build's logic script on `sdk`'s `config` <br> • Receive `baseInitialsBuilder` from the build or from `bindImage` or from a default `ripe-sdk`'s `__baseInitialsBuilder`. <br> • Receive a context function from the sdk configuration or `bindImage`, that is used to make permutations with the profiles from the current engraving. |
| Animated GIF | Oct-15-2020 17-09-07<br>Configuration example in the build: <br>Screenshot 2020-10-21 at 09 03 41|

### CODE REVIEW

**Comment by BeeMargarida** [comment]
Assigning back to me so that I can make the changes discussed:
* context variable instead of function, and ripe-commons-pluginus is reponsible for updating the context
* `runLogicP` call extracted to a function and its call in config becomes optional
* Unit tests!

**Comment by gcandal** [comment]
Closing this until we have a solid implementation on the BE.

File: src/js/base/build.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    require("./ripe");
+    // eslint-disable-next-line no-redeclare
+    var ripe = base.ripe;
+}
+
+/**
+ * Requests the logic script from the build of the current model and
+ * overrides the methods available in it.
+ */
+ripe.Ripe.prototype.setBuildLogic = async function() {
```
Comment (gcandal) [inline src/js/base/build.js:19]: loadbuildlogic

File: src/js/base/ripe.js [hunk 2]
```diff
@@ -445,6 +446,8 @@ ripe.Ripe.prototype.config = async function(brand, model, options = {}) {
         );
     }
 
+    if (this.options.useBuildLogic) await this.setBuildLogic();
```
Comment (gcandal) [inline src/js/base/ripe.js:449]: do this in parallel with setParts

File: src/js/visual/image.js [hunk 3]
```diff
@@ -544,3 +556,117 @@ ripe.Image.prototype._unregisterHandlers = function() {
...
+ * Returns the initials and profiles for the given engraving and context.
+ *
+ * @param {String} initials The value of the initials to compute the computed
+ * initials object.
+ * @param {String} engraving The value of the engraving to compute the computed
+ * initials object.
+ * @param {Element} element The DOM element to be used in the "calculus" of the
+ * final initials object.
+ */
+ripe.Image.prototype.initialsBuilder = function(initials, engraving, element) {
```
Comment (gcandal) [inline src/js/visual/image.js:570]: signature similar to server-side

**Pull request closed without merging**